inst/private/python_env_is_cygwin_like.m

```diff
@@ -0,0 +1,53 @@
+%% Copyright (C) 2022 Alex Vong
+%%
+%% This file is part of OctSymPy.
+%%
+%% OctSymPy is free software; you can redistribute it and/or modify
+%% it under the terms of the GNU General Public License as published
+%% by the Free Software Foundation; either version 3 of the License,
+%% or (at your option) any later version.
+%%
+%% This software is distributed in the hope that it will be useful,
+%% but WITHOUT ANY WARRANTY; without even the implied warranty
+%% of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
+%% the GNU General Public License for more details.
+%%
+%% You should have received a copy of the GNU General Public
+%% License along with this software; see the file COPYING.
+%% If not, see <https://www.gnu.org/licenses/>.
+
+%% -*- texinfo -*-
+%% @defun  python_env_is_cygwin_like (pyexec)
+%% Check if Python @var{pyexec} is running in a Cygwin-like POSIX environment,
+%% such as Cygwin or MSYS2.  The result is memoized to speed up subsequent
+%% calls.
+%%
+%% @seealso{cygpath}
+%% @end defun
+
+function r = python_env_is_cygwin_like (pyexec)
+  persistent python_env_is_cygwin_like_memo
+
+  if ~isempty (python_env_is_cygwin_like_memo)
+    r = python_env_is_cygwin_like_memo;
+  elseif ispc ()
```

Comment:
here I would definitely do an early return.  Then `if ispc ()`.  For me, this makes the persistent semantics clearer.

BTW, I like `_memo` for such variables!  I shall try to remember that.

Reply:
> here I would definitely do an early return. Then `if ispc ()`. For me, this makes the persistent semantics clearer.
> 
Make sense to me, it also makes the code look more symmetric.

> BTW, I like `_memo` for such variables! I shall try to remember that.
> 
The name is borrowed from [memoization](https://en.wikipedia.org/wiki/Memoization). I think I saw it somewhere as well. This is one of the reasons why [pure function](https://en.wikipedia.org/wiki/Pure_function) (and [referential transparency](https://en.wikipedia.org/wiki/Referential_transparency)) is good!